### PR TITLE
Move geo_shape disk usage test to the spatial module

### PR DESCRIPTION
**Pull request by iverase**
Adds a new SpatialDiskUsageIT test in the spatial module to test fields that the module adds.

### CODE REVIEW

File: x-pack/plugin/spatial/src/internalClusterTest/java/org/elasticsearch/xpack/spatial/SpatialDiskUsageIT.java [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+        assertThat(value, equalTo(0));
+        value = extractValue(new String[] { "test-index", "fields", "_seq_no", "points_in_bytes" }, objects);
+        assertThat(value, greaterThan(0));
+        value = extractValue(new String[] { "test-index", "fields", "_seq_no", "doc_values_in_bytes" }, objects);
+        assertThat(value, greaterThan(0));
+    }
+
+    @SuppressWarnings("unchecked")
+    private int extractValue(String[] path, Map<String, Object> objects) {
+        for (int i = 0; i < path.length - 1; i++) {
```
Comment (dnhatn) [inline x-pack/plugin/spatial/src/internalClusterTest/java/org/elasticsearch/xpack/spatial/SpatialDiskUsageIT.java:143]: Can we delegate these to `XContentMapValues#extractValue` instead?

**Review by dnhatn — APPROVED** [review]
LGTM. I left two minor comments. Thanks @iverase.

File: x-pack/plugin/spatial/src/internalClusterTest/java/org/elasticsearch/xpack/spatial/SpatialDiskUsageIT.java (hunk 1 above)
Comment (iverase) [reply x-pack/plugin/spatial/src/internalClusterTest/java/org/elasticsearch/xpack/spatial/SpatialDiskUsageIT.java]: oh yes, thanks!

File: x-pack/plugin/spatial/src/internalClusterTest/java/org/elasticsearch/xpack/spatial/SpatialDiskUsageIT.java [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+        value = extractValue(new String[] { "test-index", "fields", "_id", "stored_fields_in_bytes" }, objects);
+        assertThat(value, greaterThan(0));
+        value = extractValue(new String[] { "test-index", "fields", "_id", "points_in_bytes" }, objects);
+        assertThat(value, equalTo(0));
+        value = extractValue(new String[] { "test-index", "fields", "_id", "doc_values_in_bytes" }, objects);
+        assertThat(value, equalTo(0));
+
+        value = extractValue(new String[] { "test-index", "fields", "_seq_no", "inverted_index", "total_in_bytes" }, objects);
+        assertThat(value, equalTo(0));
+        value = extractValue(new String[] { "test-index", "fields", "_seq_no", "stored_fields_in_bytes" }, objects);
```
[the remark this answers is not in the document]
Comment (iverase) [reply x-pack/plugin/spatial/src/internalClusterTest/java/org/elasticsearch/xpack/spatial/SpatialDiskUsageIT.java]: done

**Pull request merged**